app_sidebar_collapsible.py

```diff
@@ -207,15 +238,13 @@ def update_store_uuids(start_date, end_date):
     Output("store-trips", "data"),
     Input('date-picker', 'start_date'),
     Input('date-picker', 'end_date'),
+    Input('date-picker-timezone', 'value'),
 )
-def update_store_trips(start_date, end_date):
-    if not start_date or not end_date:
-        end_date_obj = date.today()
-        start_date_obj = end_date_obj - timedelta(days=7)
-    else:
-        start_date_obj = date.fromisoformat(start_date) 
-        end_date_obj = date.fromisoformat(end_date)
-    df = query_confirmed_trips(start_date_obj, end_date_obj)
+def update_store_trips(start_date, end_date, timezone):
+    # trim the time part, leaving only date as YYYY-MM-DD
+    start_date = start_date[:10] if start_date else None
+    end_date = end_date[:10] if end_date else None
```

Comment:
this code is repeated multiple times and should be pulled out into a separate function.
This is not just for lowering the lines of code (from 2 to 1) but consolidating the calls so that if we ever change the display/parse method, we can do so in one place.

Reply:
That would be much better! I didn't see any existing files where a function like that would make sense to go in.
So to refactor, I think I will create a new file `utils/datetime_utils.py` with this new function, and maybe also the `get_ts_range` function (which I had created in `db_utils`)

Reply:
refactored